dangerzone/isolation_provider/qubes.py

```diff
@@ -84,50 +84,37 @@ def _convert(
         percentage = 0.0
 
         with open(document.input_filename, "rb") as f:
-            # TODO handle lack of memory to start qube
-            if getattr(sys, "dangerzone_dev", False):
-                # Use dz.ConvertDev RPC call instead, if we are in development mode.
-                # Basically, the change is that we also transfer the necessary Python
-                # code as a zipfile, before sending the doc that the user requested.
-                p = subprocess.Popen(
-                    ["/usr/bin/qrexec-client-vm", "@dispvm:dz-dvm", "dz.ConvertDev"],
-                    stdin=subprocess.PIPE,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE,
-                )
+            p = self.qrexec_subprocess()
+            try:
                 assert p.stdin is not None
-
-                # Send the dangerzone module first.
-                self.teleport_dz_module(p.stdin)
-
-                # Finally, send the document, as in the normal case.
                 p.stdin.write(f.read())
                 p.stdin.close()
-            else:
-                p = subprocess.Popen(
-                    ["/usr/bin/qrexec-client-vm", "@dispvm:dz-dvm", "dz.Convert"],
-                    stdin=f,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.DEVNULL,
-                )
+            except BrokenPipeError as e:
+                # Likely that qube did not have enough RAM to start
+                # in which case, it exits with code 126
+                if p.wait() == 126:
+                    # TODO: disambiguate from qrexec policy misconfigured, if possible
+                    raise errors.QubesNotEnoughRAMError()
+                else:
+                    # XXX preserve the exception since it can also have other causes
+                    # e.g.: Ctrl+C with the right timing can trigger BrokenPipeError
+                    raise e
```

Comment:
Now that I have reworked the #551  PR, I think I have a much better solution to this:

 ```suggestion
            except BrokenPipeError as e:
                    raise errors.InterruptedConversion()
```

and change the error code of `QubesNotEnoughRAMError` to 126. This way, this error will be detected at a central point (the `try ... except` clause that I introduce in this PR), and we will avoid having custom error handling logic scattered in the code.

_(minor note, when re-raising exception, it's best to use `raise` instead of `raise e`, since the former preserves the original traceback)_

Reply:
Yes, that works. I had also considered that 126 "hack" but ultimately didn't submit it because I it felt like we were mixing two distinct things. But now seeing this again and that you had a similar intuition, it might make sense.

Reply:
Yeah, I feel this asymmetry as well. We _could_ say that this error is actually returned by `qrexec`, and not by the disposable qube, just to explain why there is this different there.